### PR TITLE
Makes angle2dir() much more accurate

### PR DESCRIPTION
**Pull request by ChangelingRain**
Each return was 22.5 degrees off, which is pretty fucking far when you've only got 360 degrees to work with.

### CODE REVIEW

File: code/__HELPERS/type2type.dm [hunk 1]
```diff
@@ -125,22 +125,39 @@
...
+			return EAST
+		if(112.5 to 157.5)
+			return SOUTHEAST
+		if(157.5 to 202.5)
+			return SOUTH
+		if(202.5 to 247.5)
+			return SOUTHWEST
+		if(247.5 to 292.5)
+			return WEST
+		if(292.5 to 337.5)
```
Comment (lordpidey) [inline code/__HELPERS/type2type.dm]: Just remove the last if statement, and call `return NORTHWEST`

**Review by lordpidey — CHANGES_REQUESTED** [review]
Small optimization.

File: code/__HELPERS/type2type.dm (hunk 1 above)
Comment (ChangelingRain) [reply code/__HELPERS/type2type.dm]: This isn't actually an optimization, since it's a switch.

Comment (lordpidey) [reply code/__HELPERS/type2type.dm]: Derp, you're right.

**Comment by Giacom** [comment]
Having a math function potentially return random results sounds like a very bad idea to me. It will most likely create hard to track bugs. It should be deterministic.

**Comment by ChangelingRain** [comment]
I'll make it deterministic if you can tell me which directions should take precedence in each case.

**Comment by lordpidey** [comment]
I'd say put priority on vertical/diagnonal directions.

**Comment by optimumtact** [comment]
I decree that it will work like a southern hemisphere toilet and prefer to go left

**Comment by ChangelingRain** [comment]
How about right, instead?
Ooh, ooh, maybe it could swap which it'd do based on if world.time is odd or not.

**Comment by optimumtact** [comment]
what lord pidey said seems sensible enough all memes aside

**Pull request merged**